### PR TITLE
Adding config to publish uiautomation utilities

### PR DESCRIPTION
**Pull request by shoatman**
Adding UI Automation Utilities. 
Fixing issue where if you don't specify the local project reference will be used and this does not convey a group id.  This appears to default to the letter 's' for reasons that are still unknown to me; however I'm setting this to the correct group id when encountered.

### CODE REVIEW

File: uiautomationutilities/build.gradle [hunk 1]
```diff
@@ -44,14 +53,78 @@ android {
...
+
+                    //Iterate over the implementation dependencies (we don't want the test ones), adding a <dependency> node for each
+                    configurations.implementation.allDependencies.each {
+                        if (it.group != null && it.name != null) {
+                            def dependencyNode = dependenciesNode.appendNode('dependency')
+                            if(it.group == 's'){
+                                dependencyNode.appendNode('groupId', groupId)
+                            }else{
+                                dependencyNode.appendNode('groupId', it.group)
+                            }
```
Comment (AdamBJohnsonx) [inline uiautomationutilities/build.gradle:89]: should we supply this as a function?

File: uiautomationutilities/build.gradle [hunk 2]
```diff
@@ -44,14 +53,78 @@ android {
...
+
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "androidx.appcompat:appcompat:$rootProject.ext.appCompatVersion"
     implementation 'junit:junit:4.13'
     implementation "androidx.test.uiautomator:uiautomator:$rootProject.ext.uiAutomatorVersion"
     implementation "androidx.test:monitor:$rootProject.ext.androidxTestMonitorVersion"
     implementation "androidx.test:core:$rootProject.ext.androidxTestCoreVersion"
-    implementation project(':testutils')
+    // keyvault per product flavor
```
Comment (shahzaibj) [inline uiautomationutilities/build.gradle:118]: I'm not sure what this comment means?

File: uiautomationutilities/build.gradle (hunk 1 above)
Comment (shoatman) [reply uiautomationutilities/build.gradle:89]: Going to see if I can resolve this in a different way.

File: uiautomationutilities/build.gradle (hunk 2 above)
Comment (shoatman) [reply uiautomationutilities/build.gradle:118]: Yep.  Copy pasta

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**